fictadvisor-api/src/v2/utils/parser/RozParser.ts

```diff
@@ -272,57 +272,48 @@ export class RozParser implements Parser {
       ],
     });
 
-    const weeksCountEveryFortnight = await this.getWeeksCount(startOfEvent, Period.EVERY_FORTNIGHT);
-    const weeksCountEveryWeek = await this.getWeeksCount(startOfEvent, Period.EVERY_WEEK);
-
     if (!event) {
       await this.eventRepository.create({
         name: pair.name,
         startTime: startOfEvent,
         endTime: endOfEvent,
         period: Period.EVERY_FORTNIGHT,
+        eventsAmount: await this.getWeeksCount(startOfEvent, Period.EVERY_FORTNIGHT),
         groupId: groupId,
         lessons: {
           create: {
             disciplineTypeId: disciplineType.id,
           },
         },
-        eventInfo: {
-          createMany: {
-            data: this.getIndexesForEventInfo(0,  weeksCountEveryFortnight - 1),
-          },
-        },
       });
     } else if (
       event.startTime.getTime() === startOfEvent.getTime() - WEEK &&
       event.period === Period.EVERY_FORTNIGHT
     ) {
       await this.eventRepository.updateById(event.id, {
         period: Period.EVERY_WEEK,
-        endTime: endOfEvent,
-        eventInfo: {
-          createMany: {
-            data: this.getIndexesForEventInfo(weeksCountEveryFortnight, weeksCountEveryWeek),
-          },
-        },
+        eventsAmount: await this.getWeeksCount(startOfEvent, Period.EVERY_WEEK),
       });
     }
   }
 
-  getIndexesForEventInfo (startIndex: number, endIndex: number) {
-    return Array.from({
-      length: endIndex - startIndex + 1,
-    }, (_, i) => ({
-      number: i + startIndex,
-    }));
+  async getWeeksCount (startOfEvent: Date, period: Period) {
+    const { startDate, endDate } = await this.dateService.getCurrentSemester();
+    const lastWeek = Math.ceil((endDate.getTime() - startDate.getTime() - FORTNITE)/ WEEK);
+    const weeks = period === Period.EVERY_FORTNIGHT ? lastWeek/2 : lastWeek;
+    if (this.isInteger(weeks)) return weeks;
+    const startWeek = Math.ceil((startOfEvent.getTime() - startDate.getTime()) / WEEK);
+    return this.weeksForEveryFortnight(startWeek, weeks);
   }
 
-  async getWeeksCount (indexStartDate: Date, period: Period) {
-    const { endDate } = await this.dateService.getCurrentSemester();
-
-    const difference = Math.ceil((endDate.getTime() - FORTNITE - indexStartDate.getTime()) / DAY);
-    const divider = period === Period.EVERY_FORTNIGHT ? 14 : 7;
+  weeksForEveryFortnight (startWeek: number, weeks: number) {
```

Comment:
you need to check whether this method is essential at all. i think there is no need to check this.

Reply:
This method is used when the number of weeks in the semester is odd. If this cannot be the case, then this method is not needed